src/Relude/Extra/Validation.hs

```diff
@@ -0,0 +1,180 @@
+{-# LANGUAGE CPP #-}
+{-# LANGUAGE InstanceSigs #-}
+{-# LANGUAGE LambdaCase #-}
+
+{- |
+Copyright: (c) 2014 Chris Allen, Edward Kmett
+           (c) 2018 Kowainik
+License:   MIT
+Maintainer: Kowainik <xrom.xkov@gmail.com>
+
+Monoidal 'Validation' sibling to 'Either'.
+-}
+
+module Relude.Extra.Validation
+       ( Validation(..)
+       , validationToEither
+       , eitherToValidation
+       ) where
+
+import Relude
+
+-- $setup
+-- >>> :set -XTypeApplications -XOverloadedStrings
+
+-- | 'Validation' is 'Either' with a Left that is a 'Monoid'
+data Validation e a
+    = Failure e
+    | Success a
+    deriving (Eq, Ord, Show)
+
+instance Functor (Validation e) where
+    fmap :: (a -> b) -> Validation e a -> Validation e b
+    fmap _ (Failure e) = Failure e
+    fmap f (Success a) = Success (f a)
+
+    (<$) :: a -> Validation e b -> Validation e a
+    x <$ Success _ = Success x
+    _ <$ Failure e = Failure e
+
+    {-# INLINE fmap #-}
+    {-# INLINE (<$) #-}
+
+-- | Examples:
```

Comment:
Maybe it's a good idea to write this as `==== __Examples__` to make them expandable 🤔 
Not sure how it will look like.

Also, IMO, for multiline haddock comments it's better to use: the following style

```haskell
{- |
-}
```

so you won't need to add `-- ` at the beginning of each line.

Reply:
The proposed expandable formatting works, but looks like this in Firefox and Chrome:

![screenshot from 2018-10-16 06-41-50](https://user-images.githubusercontent.com/180308/47015391-72c61200-d112-11e8-9f6d-cb78627d199f.png)

It currently looks like this:

![screenshot from 2018-10-16 06-41-43](https://user-images.githubusercontent.com/180308/47015405-85d8e200-d112-11e8-89dc-c5b504810ae9.png)

:thinking:

Reply:
I think that non-expandable is better :slightly_smiling_face: